coral/src/app/features/topics/details/schema/TopicDetailsSchema.tsx

```diff
@@ -25,18 +35,68 @@ function TopicDetailsSchema() {
       allSchemaVersions = [],
       latestVersion,
       schemaDetailsPerEnv,
+      schemaPromotionDetails,
     },
     setSchemaVersion,
     topicOverview: { topicInfoList },
   } = useTopicDetails();
+  const [showSchemaPromotionModal, setShowSchemaPromotionModal] =
+    useState(false);
+  const [errorMessage, setErrorMessage] = useState("");
 
   const isTopicOwner = topicInfoList[0].topicOwner;
   const noSchema =
-    allSchemaVersions.length === 0 || schemaDetailsPerEnv === undefined;
+    allSchemaVersions.length === 0 ||
+    schemaDetailsPerEnv === undefined ||
+    schemaPromotionDetails === undefined;
 
-  function promoteSchema() {
-    console.log("dummy function");
-  }
+  const toast = useToast();
+
+  const { mutate: promoteSchema, isLoading: promoteSchemaIsLoading } =
+    useMutation(
+      ({
+        forceRegister = false,
+        remarks = "",
+      }: Pick<PromoteSchemaPayload, "forceRegister" | "remarks">) => {
+        if (noSchema) {
+          throw new Error("No schema available");
+        }
+
+        const { targetEnvId, sourceEnv } =
+          schemaPromotionDetails[schemaDetailsPerEnv.env] ?? {};
+
+        if (targetEnvId === undefined || sourceEnv === undefined) {
+          throw new Error("No promotion details available available");
+        }
+
+        return promoteSchemaRequest({
+          targetEnvironment: targetEnvId,
+          sourceEnvironment: sourceEnv,
+          topicName,
+          schemaVersion: String(schemaDetailsPerEnv.version),
+          schemaFull: schemaDetailsPerEnv.content,
+          forceRegister,
+          // @TODO: get the appname form topicOverview, probably needs to be added in the backend?
```

Comment:
Is the app name even variable? In `createSchemaRequest` we used it hard set after sync with backend, because it's always `app` anyways.

Reply:
Yes, we hardcode it everywhere currently, but I think it's there to enable upcoming features? @muralibasani , @aindriu-aiven can you confirm?

Reply:
I mean, either way it's revertible! Just wanted to mention that in the create endpoint, we set it directly so the components etc don't know about it yet.

Reply:
I'm not sure if appName will be used in another form in the future at the moment it is hardcoded everywhere it is used.

Reply:
Should we update this part then to match how we handled it in creating a new schema? Or at least update the comment, because it's not really a todo now.

Reply:
I think I like it in this form because it doesn't force us to create an artificial `Omit` type in `domain` so we can be allowed to pass a payload without `appName`  in `app`. And it looks weird enough that we won't forget to address the issue eventually.

Reply:
I think having it contained in `domain` keeps the "app" code cleaner and a bit easier to understand. For us as core maintainer this is not really important, because we've the internal understanding, but I'm also thinking about potential outside contributors, who will more likely stumble or it here than in `domain`. And that's what `domain` is for! Keep things that need or profit from implicit knowledge contained and therefor create a clearer domain model for coral as an interface.

Also, in my experience, it does not really make a difference if there's a weird looking thing here or in `domain` with a TODO comment - we won't think about it unless we create an issue for it 😅  (I'm including myself here, it's just how our brain tend to work)

I'm not stuck on this, so feel free to move it or not :D But at least the comment then should be up to date.

Reply:
After trying it out, it does look better ^^